### PR TITLE
search --limit compatible with docker

### PR DESCRIPTION
**Pull request by QiWang19**
Check --limit range and update --limit manpage explanation.

Signed-off-by: Qi Wang <qiwan@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM
But it looks like you have broken the tests.  Also do we know what the default is?  Should be mentioned in the the man page.

File: docs/source/markdown/podman-search.1.md [hunk 1]
```diff
@@ -59,7 +59,7 @@ Valid placeholders for the Go template are listed below:
 
 **--limit**=*limit*
 
-Limit the number of results
+Limit the number of results. This value could be in the range between 1 and 100.
```
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-search.1.md:62]: nit could -> can

**Comment by TomSweeneyRedHat** [comment]
The other part of this problem that's not solved here is if there are 200 entries in the registry and the registry returns only 100 due to the limit set by the registry itself, regardless of the limit option being used or not used by the user.  So if there are 200 entries in the registry and the user asks for all but only 100 are shown, we need to tell them at the end of the output.  Similar to what Docker does and Derrick put an example in the BZ.

**Comment by TomSweeneyRedHat** [comment]
And fwiw, I'm fine with merging this in the near term and then doing a separate PR for the other warning message.

**Comment by TomSweeneyRedHat** [comment]
LGTM

**Comment by vrothberg** [comment]
@TomSweeneyRedHat @QiWang19 @rhatdan what was the BZ? I just hit this issue (working on another BZ) and do not understand this limit.

**Comment by TomSweeneyRedHat** [comment]
@vrothberg the BZ is https://bugzilla.redhat.com/show_bug.cgi?id=1827794

**Comment by vrothberg** [comment]
I don't understand the hard limit of 100 if it's up to registries to decide.

**Comment by vrothberg** [comment]
@mtrmac WDYT?

**Comment by mtrmac** [comment]
Yeah, that’s not what the bug asked for. On the contrary, if anything, the user should be able to disable the default 25 limit to get _all_ results without any limit.

**Pull request merged**